src/apphosting/githubConnections.ts

```diff
@@ -180,6 +195,107 @@ async function createFullyInstalledConnection(
   return conn;
 }
 
+async function manageInstallation(connection: devConnect.Connection): Promise<void> {
+  utils.logBullet(
+    "Manage the Firebase App Hosting GitHub app to enable access to GitHub repositories",
+  );
+  const targetUri = connection.githubConfig?.installationUri;
+  if (!targetUri) {
+    throw new Error("No installation given to manage");
+  }
+
+  utils.logBullet(targetUri);
+  await utils.openInBrowser(targetUri);
+  await promptOnce({
+    type: "input",
+    message:
+      "Press Enter once you have installed or configured the Firebase App Hosting GitHub app to access your GitHub repo.",
+  });
+}
+
+export async function getConnectionForInstallation(
+  projectId: string,
+  location: string,
+  installationId: string,
+): Promise<devConnect.Connection | null> {
+  const connections = await listAppHostingConnections(projectId, location);
+  const connectionsMatchingInstallation = connections.filter(
+    (conn) => conn.githubConfig?.appInstallationId === installationId,
+  );
+  if (connectionsMatchingInstallation.length === 0) {
+    return null;
+  }
+
+  if (connectionsMatchingInstallation.length > 1) {
+    // return the oldest connection
+    const sorted = devConnect.sortConnectionsByCreateTime(connectionsMatchingInstallation);
+    return sorted[0];
+  }
```

Comment:
q: is there a situation where this is a valid state?

(if not, maybe we can comment on this being invalid but we'll go ahead and handle the situation gracefully by picking the oldest connection)

Reply:
No I don't believe so. With the changes in this PR we want to ensure there isn't more than one connection per installation. But in the Console and CLI currently we are creating a connection and then choosing an installation, which makes it possible for there to be more than one connection for the same installation. So we should handle it gracefully.

I'll add a comment here.